include/vkwrap/vkwrap/sampler.h

```diff
@@ -0,0 +1,186 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include "vkwrap/core.h"
+#include "vkwrap/mman.h"
+#include "vkwrap/utils.h"
+
+#include "utils/patchable.h"
+
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/conversion.hpp>
+
+#include <functional>
+#include <optional>
+#include <utility>
+
+namespace vkwrap
+{
+
+class Sampler : private vk::UniqueSampler
+{
+
+  private:
+    using Base = vk::UniqueSampler;
+
+  public:
+    Sampler( vk::UniqueSampler unique_sampler )
+        : Base{ std::move( unique_sampler ) }
+    {
+    }
+
+    using Base::operator bool;
+    using Base::operator->;
+    using Base::get;
+
+}; // class Sampler
+
+class SamplerBuilder
+{
+
+  public:
+    // clang-format off
+    PATCHABLE_DEFINE_STRUCT( 
+        SamplerPartialInfo,
+        ( std::optional<vk::Filter>,             mag_filter               ),
+        ( std::optional<vk::Filter>,             min_filter               ),
+        ( std::optional<vk::SamplerAddressMode>, address_mode_u           ),
+        ( std::optional<vk::SamplerAddressMode>, address_mode_v           ),
+        ( std::optional<vk::SamplerAddressMode>, address_mode_w           ),
+        ( std::optional<vk::Bool32>,             anisotropy_enable        ),
+        ( std::optional<vk::CompareOp>,          compare_op               ),
+        ( std::optional<vk::BorderColor>,        border_color             ),
+        ( std::optional<vk::Bool32>,             unnormalized_coordinates )
+    );
+    // clang-format on
+
+    using Setter = std::function<void( SamplerPartialInfo& )>;
+
+  private:
+    // clang-format off
+    Setter m_setter = []( auto& ){};
+    // clang-format on
+
+    SamplerPartialInfo m_partial;
+
+    SamplerPartialInfo makePartialInfo() const
+    {
+        SamplerPartialInfo partial{};
+
+        presetter( partial );
+        m_setter( partial );
+        partial.patchWith( m_partial );
+
+        return partial;
+    } // makePartialInfo
+
+    static float getMaxAnisotropy( vk::PhysicalDevice physical_device )
+    {
+        return physical_device.getProperties().limits.maxSamplerAnisotropy;
+    } // getMaxAnisotropy
+
+    static constexpr vk::SamplerCreateInfo k_sampler_initial_create_info{
+        // We don't use these specific fields.
+        .pNext = {},
+        .flags = {},
+
+        // We use images with only one mipmap.
+        .mipmapMode = vk::SamplerMipmapMode::eLinear,
+        .mipLodBias = 0.0f,
+
+        // compareOp would define behaviour of comparison.
+        .compareEnable = VK_TRUE,
+
+        // We use images with only one mipmap.
+        .minLod = 0.0f,
+        .maxLod = 0.0f,
+    };
+
+  public:
+    // clang-format off
+    static inline Setter presetter = []( auto& ){};
```

Comment:
Why static inline? This can be a static constexpr variable without storage

Reply:
Purpose of the `presetter` - give to user an opportunity to configure preset of `SamplerPartialnfo`. Of course, user can change it.